### ISSUE TITLE
Report exception is thrown in caller method

### ISSUE THREAD

**Issue by davidmpaz** [kalessil/phpinspectionsea#694]
Hi,

I know from couple issue the inspections for exceptions is complicated. But as a note for future lets have this issue created.

Problem:
I found my self more and more writing functions without doc blocks at all. PS doesn't warn me that I am not handling a **possible** thrown exception in my caller method when both of them are not annotated with proper doc blocks.

```
// file1.php
public function getModelById(int $id): SomeModelInterface
{
    $entity = $this->someService->getMeTheEntity();

    if (!$entity instanceof SomeRessource) {
        throw new NotValidException('Model with id: ' . $id);
    }

    return $this->getModelFromEntity($entity);
}

// file2.php
public function handleGetModel(Request $request): Response
{
    $id = $this->getIdFromRequest($request);
    $data = $this->service->getModelById($id)->toHash();

    $response = new JsonView($data);
    return $response->getResponse();
}
```

Shall I report this to PhpStorm guys?

Cheers

**Comment by tomzx** [comment]
It seems there's a bug with recent changes in PHPStorm which makes it lose the ability to identify the SPL exceptions for some reason. See https://youtrack.jetbrains.com/issue/WI-30775.

**Comment by davidmpaz** [comment]
Hi @tomzx,

thanks for answering. I think it goes a bit further. It happens also with user land Exceptions. The result I am looking for is that, there should not be need to annotate the method with `@throw` tag at all in order to PS recognize that there is not handled Exceptions to take into account.

That's why the question: Shall I report/request to PS guys?  I am not even sure that is a bug, maybe is a missing feature. Dunno honestly.

cheers

**Comment by kalessil** [comment]
At some point, we'll drop the inspection on our end (when PS team catches up on our features), so it worth reporting an issue to them.

On our end, I'll check if I can add this to Ultimate (inspection code is a mess frankly speaking). Hopefully, I manage to add this check.

**Comment by davidmpaz** [comment]
Reported in PS tracker: https://youtrack.jetbrains.com/issue/WI-39257

**Comment by kalessil** [comment]
@davidmpaz : happens from time to time =) Then, let's keep building own casino with blackjack and ...

**Comment by kalessil** [comment]
I added basic support for the not-annotated methods. @davidmpaz : will this be sufficient or we need to do extra work here https://github.com/kalessil/phpinspectionsea/commit/30117d2648e4c8ca978db2ec584f2cb8e0a30083#diff-a6a3ca30113e79f6797fa06ebbcd7386 ?

**Comment by kalessil** [comment]
Re-opening: my change doesn't check try-statements
